src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,38 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE_LIMIT = 18;
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegisterException {
+        User user1 = storageDao.get(user.getLogin());
+        if (user.getLogin() == null) {
+            throw new RegisterException("Login can't be null");
+        }
+        if (user.equals(user1)) {
+            throw new RegisterException("This user is already in the database");
+        }
+        if (user.getLogin().isEmpty()) {
+            throw new RegisterException("Login can't be empty");
+        }
+        if (user.getPassword() == null) {
+            throw new RegisterException("Password can't be null");
+        }
+        if (user.getPassword().isEmpty()) {
+            throw new RegisterException("Password can't be empty");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegisterException("The length of the password must be at least 6 characters");
+        }
+        if (user.getAge() == null) {
+            throw new RegisterException("Age can't be null");
+        }
+        if (user.getAge() < MIN_AGE_LIMIT) {
+            throw new RegisterException("The user is under 18 years old");
+        }
```

Comment:
Don’t create one big if with all conditions. Better to split one big if into several small ones.

Reply:
I did so. if not, specify where exactly. thank you